clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -4117,11 +4117,10 @@ static void maybeEmitPipeStorageMetadata(const VarDecl *D,
   if (!PipeTy->isStructureType())
     return;
 
-  if (auto *IOAttr = D->getAttr<SYCLIntelPipeIOAttr>()) {
-    Optional<llvm::APSInt> ID =
-        IOAttr->getID()->getIntegerConstantExpr(D->getASTContext());
+  if (const auto *IOAttr = D->getAttr<SYCLIntelPipeIOAttr>()) {
+    const auto *CE = cast<ConstantExpr>(IOAttr->getID());
```

Comment:
Can you explain why we need this change?

Reply:
This patch stores expression as ConstantExpr instead of IntegerConstantExpr. So i have made same changes in semantic as well as codegen part.